doc/Neighbor/Neighbor_Refresh_HLD.md

```diff
@@ -0,0 +1,321 @@
+
+# Neighbor Refresh HLD
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [Revision](#revision)
+  * [Definition/Abbreviation](#definitionabbreviation)
+  * [About This Manual](#about-this-manual)
+  * [1 Requirement Overview](#1-requirement-overview)
+     * [1.1 Functional Requirements](#11-functional-requirements)
+     * [1.2 Configuration and Management Requirements](#12-configuration-and-management-requirements)
+     * [1.3 Scalability Requirements](#13-scalability-requirements)
+     * [1.4 Warm Boot Requirements](#14-warm-boot-requirements)
+     * [1.5 Limitiations](#15-limitations)
+  * [2 Functionality](#2-functionality)
+     * [2.1 Functional Description](#21-functional-description)
+  * [3 Design](#3-design)
+     * [3.1 Overview](#31-overview)
+     * [3.2 DB Changes](#32-db-changes)
+     * [3.3 Switch State Service Design](#33-switch-state-service-design)
+     * [3.4 SyncD](#34-syncd)
+     * [3.5 SAI](#35-sai)
+     * [3.6 CLI](#36-cli)
+  * [4 Flow Diagrams](#4-flow-diagrams)
+  * [5 Warm Boot Support](#5-warm-boot-support)
+  * [6 Unit Test](#6-unit-test)
+
+
+# Revision
+| Rev  |    Date     |   Author     | Change Description |
+| :--: | :---------: | :---------:  | ------------------ |
+| 0.1  | 25-Jul-2022 | Laveen       | Initial version    |
+|      |             |              |                    |
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term** | **Meaning**                                       |
+| -------- | ------------------------------------------------- |
+| ARP      | Address Resolution Protocol                       |
+| ND       | Neighbor Discovery                                |
+| IP       | Internet Protocol                                 |
+| L3       | Layer 3                                           |
+| RIF      | Routing Interface                                 |
+| VRF      | Virtual Routing and Forwarding                    |
+| SVI      | Switch Virtual Interface                          |
+
+# About This Manual
+This document describes the high level description of neighbor refresh in SONiC.
+
+# 1 Requirement Overview
+
+Refresh ARP/ND before their respective timer or MAC ageout.
+
+SONiC depends upon the Linux kernel to manage the ARP/ND tables. SONiC then listens to ARP/ND events from the kernel and synchronizes the hardware as required. Some of the limitations with this are as follows,
+
+- The kernel does not "see" the routed (in HW) through-traffic, and so cannot update its "hit bits" accordingly. Therefore the kernel may age out an entry that is still in use.
+- The kernel does not "see" the HW MAC aging process, and so does not know that a MAC address associated with an ARP/ND entry has been aged out, and so does not refresh it. This can result in traffic black holes for a "quiet" neighbor (i.e. one that does not transmit much) / unidirectional traffic streams.
```

Comment:
What happens when the MAC is aged out in the kernel? do we refresh the ARP/Nbr to re-learn the MAC?

Reply:
It is recommended to have the same MAC age out time in kernel and the HW.

Nbrmgrd will be listing to state FDB DB changes, if there is no change in state FDB DB for kernel ageout, ARP/Nbr will not be refreshed.